### PR TITLE
[state sync] add tests for executor proxy interface

### PR DESCRIPTION
**Pull request by christinacelee**
## Motivation

more tests to increase test coverage for `executor_proxy` under state sync

## Test Plan

Local test coverage with new test is 85% for `executor_proxy.rs`

### CODE REVIEW

**Review by sausagee — APPROVED** [review]
Thanks for getting this in!

File: state-synchronizer/src/tests/on_chain_config_tests.rs [hunk 1]
```diff
@@ -199,4 +199,77 @@ fn test_on_chain_config_pub_sub() {
...
+    // Case 3: test failed on-chain reconfig publish throws error //
+    ////////////////////////////////////////////////////////////////
+    // test error throwing on failed reconfig publish
+    drop(reconfig_receiver);
+
+    // Create a dummy block prologue transaction that will bump the timer.
+    let txn6 = encode_block_prologue_script(gen_block_metadata(3, validator_account));
+    let txn7 = get_test_signed_transaction(
+        genesis_account,
+        /* sequence_number = */ 3,
```
Comment (davidiw) [inline state-synchronizer/src/tests/on_chain_config_tests.rs:213]: kinda surprised we don't use a variable for this instead and increment it afterward

File: state-synchronizer/src/tests/on_chain_config_tests.rs [hunk 2]
```diff
@@ -199,4 +199,77 @@ fn test_on_chain_config_pub_sub() {
...
+        .commit_blocks(vec![block3_id], ledger_info_with_sigs)
+        .unwrap();
+    assert!(
+        !reconfig_events.is_empty(),
+        "expected reconfig events from executor commit"
+    );
+
+    assert!(executor_proxy
+        .publish_on_chain_config_updates(reconfig_events)
+        .is_err());
```
Comment (davidiw) [inline state-synchronizer/src/tests/on_chain_config_tests.rs:245]: this is what you're testing for? maybe a comment?

File: state-synchronizer/src/tests/on_chain_config_tests.rs [hunk 3]
```diff
@@ -199,4 +199,77 @@ fn test_on_chain_config_pub_sub() {
...
+    assert_eq!(epoch_li, ledger_info_with_sigs_epoch_2);
+
+    // epoch 1 ended at version 2
+    let epoch_li = executor_proxy.get_epoch_ending_ledger_info(2).unwrap();
+    assert_eq!(epoch_li, ledger_info_with_sigs_epoch_1);
+    // epoch 2 ended at version 5
+    let epoch_li = executor_proxy.get_epoch_ending_ledger_info(5).unwrap();
+    assert_eq!(epoch_li, ledger_info_with_sigs_epoch_2);
+
+    assert!(executor_proxy.get_epoch_ending_ledger_info(3).is_err());
```
Comment (davidiw) [inline state-synchronizer/src/tests/on_chain_config_tests.rs:274]: at this point all these values look magical, it isn't clear where they're coming from. a 220 line test function is ... a lot to swallow... is there anyway we can refactor these into 3 tests?

Comment (christinacelee) [reply state-synchronizer/src/tests/on_chain_config_tests.rs:274]: all the test cases build up on each other, and reconstructing the DB state tested against is pretty expensive LOC-wise and runtime wise... :/

**Pull request closed without merging**